docs/rfcs/018-revisit-batches.md

```diff
@@ -0,0 +1,68 @@
+# Revisit Batches
+
+See [#491](https://github.com/dlcs/protagonist/issues/491) for original ticket outlining this issue.
+
+Currently a "Batch" is a loose and transient grouping of images. An image can belong to 1 batch at any given time, no history is maintained. This worked well for the initial use-case of bulk ingestion of images and tracking process for things like progress bars. However for more complex scenarios this connection is lacking and improving it 
+
+This RFC outlines proposal for improving batches, this will allow us to:
+* Get a list of assets in a batch, not just the current batch.
+* Extend named query support to filter on batchId(s)
+* Avoid issues with double counting completed or failed assets
+* Track which batches an images has been ingested in
+* Identify which assets failed within a batch
+* Reingest any failed assets in a batch (see [#402](https://github.com/dlcs/protagonist/issues/402))
+
+Note that not all of the above points are addressed here, some will need new API endpoints to be added building on the change outlines here.
+
+## Proposal
+
+The proposal is to introduce a new table, `BatchAssets`. This will track the assetIds per batch, maintaining the history of batch:asset. 
+
+The following is the proposed schema:
+
+| Column   | Type        | Description                                    |
+| -------- | ----------- | ---------------------------------------------- |
+| BatchId  | int         | Batch identifier                               |
+| AssetId  | text        | Asset identifier                               |
+| Status   | int         | Values of waiting/completed/error map to enum. |
+| Finished | timestamptz | Date when asset finished                       |

```

Comment:
Would it be worth having an `Error` or mappable `ErrorCode` field here?

Reply:
Good shout, added